crypto/evp/scrypt.c

```diff
@@ -139,7 +139,7 @@ static void scryptROMix(unsigned char *B, uint64_t r, uint64_t N,
         *pV = *pB++;
         *pV |= *pB++ << 8;
         *pV |= *pB++ << 16;
-        *pV |= *pB++ << 24;
+        *pV |= (uint32_t)*pB++ << 24;
```

Comment:
As for "it doesn't seem to affect the result." The statement can be made as broad as "it doesn't affect the result on any of the supported platforms," because all of them are two's complement. As for other shifts, a number of similar casts were already made to appease clang sanitizer since a while back. The fix is fine.